application/cms/page_service.py

```diff
@@ -299,10 +299,13 @@ def create_measure_version(self, measure_version, update_type, user):
         new_version.latest = True
 
         new_version.uploads = []
-        for upload in measure_version.uploads:
-            new_upload = upload.copy()
-            new_upload.guid = create_guid(upload.file_name)
-            new_version.uploads.append(new_upload)
+
+        # We don't copy uploads for major updates, as major updates should always have new data
+        if update_type != NewVersionType.MAJOR_UPDATE:
```

Comment:
This will also copy across uploads for duplicated measure versions. I guess this is fine (possibly even ideal) while the 'copy measure' functionality is only used internally for testing/training. And there's probably no situation where we'd ever want to use 'copy measure' for a legitimate measure, I suppose?

Reply:
Yeah I thought about this and I reckon the copy should copy everything.  We've literally never used it, and seems unlikely we ever will.  But if we do then having the files seems good.

Reply:
Super minor point, but it might be slightly easier to understand as `if update_type == NewVersionType.MINOR_UPDATE`, rather than a negative condition? The only other option is `NewVersionType.NEW_MEASURE`, which is already excluded from the parent if/else condition, but logically only minor updates would do the copying? No biggie though.